Stratis.Bitcoin/Utilities/AsyncLock.cs

```diff
@@ -1,131 +1,47 @@
 ﻿using System;
-using System.Collections;
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
-using System.Linq;
+using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
-using NBitcoin;
 
 namespace Stratis.Bitcoin.Utilities
 {
-    /// <summary>
-    /// An async reader writer lock for concurrent and exclusive work.
-    /// </summary>
-    public interface IAsyncLock
+    // https://www.hanselman.com/blog/ComparingTwoTechniquesInNETAsynchronousCoordinationPrimitives.aspx
```

Comment:
This whole new file lacks comments, pls make new code conforming with the commenting policy. See the new `ISchedulerLock` for inspiration. Also uses `var` against the code policy.

Reply:
Yes please add comments this will be very useful